src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -99,23 +149,155 @@ const SKUpdate& SKNMEA2000Parser::parse128267(const SKSourceInput& input, const
 
     update->setEnvironmentDepthBelowTransducer(depthBelowTransducer);
 
-    if (!N2kIsNA(offset)) {
-      // When offset is negative, it's the distance between transducer and keel
-      if (offset < 0) {
-        update->setEnvironmentDepthTransducerToKeel(offset * -1);
-        update->setEnvironmentDepthBelowKeel(depthBelowTransducer + offset);
-      }
-      else if (offset > 0) {
-        update->setEnvironmentDepthSurfaceToTransducer(offset);
-        update->setEnvironmentDepthBelowSurface(depthBelowTransducer + offset);
+    if (N2kIsNA(offset)) {
+      // When no offset then offset should be Zero
+      offset = 0;
+    }
+    // When offset is negative, it's the distance between transducer and keel
+    if (offset < 0) {
+      update->setEnvironmentDepthTransducerToKeel(offset * -1);
+      update->setEnvironmentDepthBelowKeel(depthBelowTransducer + offset);
+    }
+    else if (offset > 0) {
+      update->setEnvironmentDepthSurfaceToTransducer(offset);
+      update->setEnvironmentDepthBelowSurface(depthBelowTransducer + offset);
+    }
+
+    _sku = update;
+    return *_sku;
+  }
+  else {
+    DEBUG("Unable to parse N2kMsg with PGN %i", msg.PGN);
+    return _invalidSku;
+  }
+}
+
+// *****************************************************************************
+//   PGN 130306 W I N D
+//    tN2kWindReference:
+//      N2kWind_True_North=0,
+//      N2kWind_Magnetic=1,
+//      N2kWind_Apprent=2,
+//      N2kWind_True_boat=3
+// *****************************************************************************
+const SKUpdate& SKNMEA2000Parser::parse130306(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;
+  double windSpeed;     // Knots
+  double windAngle;     // in Radians 0... 2*pi
+  tN2kWindReference windReference;
+
+  if (ParseN2kPGN130306(msg,sid,windSpeed,windAngle,windReference) ) {
+    SKUpdateStatic<2> *update = new SKUpdateStatic<2>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+
+    if (!N2kIsNA( windAngle) && !N2kIsNA( windSpeed)) {
+      if (windReference == N2kWind_Apprent) {
+        // -> m/s (Meters per second)
+        update->setEnvironmentWindSpeedApparent(SKKnotToMs(windSpeed));
```

Comment:
This is incorrect. The NMEA2000 source code indicates that this should already be in m/s. See https://github.com/ttlappalainen/NMEA2000/blob/master/src/N2kMessages.h#L1204.